### PR TITLE
Remove public key from trunk model permissions

### PR DESCRIPTION
**Pull request by jmorel**
Companion PR https://github.com/SubstraFoundation/substra-backend/pull/188

Remove public key from out composite traintuple specs.

### CODE REVIEW

**Comment by jmorel** [comment]
@AurelienGasser @samlesu could you re-review this PR ? I added a test making sure you cannot force public permissions for a composite traintuple trunk.

File: tests/test_permissions.py [hunk 1]
```diff
@@ -314,3 +314,30 @@ def test_permissions_denied_head_model_process(factory, client_1, client_2):
     )
     with pytest.raises(substra.exceptions.InvalidRequest):
         client_2.add_composite_traintuple(spec)
+
+
+def test_permission_public_trunk(factory, client):
```
Comment (samlesu) [inline tests/test_permissions.py:319]: It is not really clear what's the purpose of this test. Adding a docstring/comments would be IMO a good idea.

**Comment by jmorel** [comment]
@samlesu done, I moved it to the system tests and added an explanation.

**Comment by samlesu** [comment]
@jmorel thanks!

**Pull request merged**